readme.md

```diff
@@ -8,80 +8,111 @@
 [![Backers][backers-badge]][collective]
 [![Chat][chat-badge]][chat]
 
-[**rehype**][rehype] plugin to transform to [**React**][react].
+**[rehype][]** plugin to compile HTML to React nodes.
+
+## Contents
+
+*   [What is this?](#what-is-this)
+*   [When should I use this?](#when-should-i-use-this)
+*   [Install](#install)
+*   [Use](#use)
+*   [API](#api)
+    *   [`unified().use(rehypeReact, options)`](#unifieduserehypereact-options)
+*   [Types](#types)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Related](#related)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## What is this?
+
+This package is a [unified][] ([rehype][]) plugin that compiles HTML (hast) to
+React nodes (the virtual DOM that React uses).
+
+**unified** is a project that transforms content with abstract syntax trees
+(ASTs).
+**rehype** adds support for HTML to unified.
+**hast** is the HTML AST that rehype uses.
+This is a rehype plugin that adds a compiler to compile hast to React nodes.
+
+## When should I use this?
+
+This plugin adds a compiler for rehype, which means that it turns the final
+HTML (hast) syntax tree into something else (in this case, a React node).
+It’s useful when you’re already using unified (whether remark or rehype) or are
+open to learning about ASTs (they’re powerful!) and want to render content in
+your React app.
+
+If you’re not familiar with unified, then [`react-markdown`][react-markdown]
+might be a better fit.
+You can also use [`react-remark`][react-remark] instead, which is somewhere
+between `rehype-react` and `react-markdown`, as it does more that the former and
+is more modern than the latter, and also a good alternative.
```

Comment:
Any tangible reason we could mention to use one over the other? "More modern" is quite vague for instance